common/src/main/java/com/github/khanshoaib3/minecraft_access/features/read_crosshair/CrosshairNarrator.java

```diff
@@ -0,0 +1,31 @@
+package com.github.khanshoaib3.minecraft_access.features.read_crosshair;
+
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.util.hit.HitResult;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface CrosshairNarrator {
+    @NotNull HitResult rayCast();
+
+    /**
+     * Determines weather the player is looking at the same block or entity.
+     * {@link #narrate(ClientWorld, boolean)} will only be called
+     * if this method returns a different non-null value than its previous call.
+     * A value of null indicates that the player is not currently looking at any block
+     * and thus {@link #narrate(ClientWorld, boolean)} should not be called.
+     * @param world The current world the player is in
+     * @param speakSide Weather the block side the player is pointing at should be narrated
```

Comment:
Typo: Weather  -> Whether (might be made by me)

Reply:
Oopsie, ill fix that in the morning. BTW, when you make a PR review comment there is a button above the comment box to make a suggestion which I can then accept with a simple push of a button.

Reply:
> there is a button above the comment box to make a suggestion which I can then accept with a simple push of a button.

Ok, I'll try to find it in Braille's PR... let me read the GitHub doc...